### ISSUE TITLE
Consider adding support for injecting a custom ObjectMapper bean

### ISSUE THREAD

**Issue by vpavic** [spring-projects/spring-hateoas#821]
To my knowledge, currently it isn't possible to inject a custom, Spring HATEOAS dedicated, `ObjectMapper` bean into `ConverterRegisteringWebMvcConfigurer` as it will either work with the primary instance, or create new one:

https://github.com/spring-projects/spring-hateoas/blob/9131a82b9070187b3c3dbb07fd8a7dd244bddd18/src/main/java/org/springframework/hateoas/config/ConverterRegisteringWebMvcConfigurer.java#L105

It would be nice to have the capability to configure Spring HATEOAS to work with a non-primary `ObjectMapper` bean as in quite a few non-trivial apps the `ObjectMapper` will be used in different places and therefore with different configuration requirements.

For example, in Spring Session we have qualifiers that enable users to provide their own, Spring Session dedicated, bean definitions for infrastructure related components. As an example, see:

- [`RedisHttpSessionConfiguration`](https://github.com/spring-projects/spring-session/blob/0c9fbedd05517036c0b17184eaf42fba8586fe23/spring-session-data-redis/src/main/java/org/springframework/session/data/redis/config/annotation/web/http/RedisHttpSessionConfiguration.java#L180-L215) - allows injection of Spring Session specific `RedisConnectionFactory`, `RedisSerializer`, `Executor`.
- [`JdbcHttpSessionConfiguration`](https://github.com/spring-projects/spring-session/blob/0c9fbedd05517036c0b17184eaf42fba8586fe23/spring-session-jdbc/src/main/java/org/springframework/session/jdbc/config/annotation/web/http/JdbcHttpSessionConfiguration.java#L149-L175) - allows injection of Spring Session specific `DataSource`, `LobHandler`, `ConversionService`.

It would be nice if Spring HATEOAS would provide similar configuration capabilities. I'm working on a project where we're adding a new version of API (HAL based) but still have to support the previous API for quite some time - it would be nice if we could have a simple way of hooking a dedicated `ObjectMapper` instance into Spring HATEOAS config. At the moment we're using a `BeanPostProcessor` in a similar manner like [described here](https://github.com/spring-projects/spring-hateoas/issues/263#issuecomment-463392669), which obviously isn't ideal.

**Comment by vpavic** [comment]
Any feedback on this? Thanks.

**Comment by odrotbohm** [comment]
In #1382 we introduced callbacks on `HalConfiguration` and `HalFormsConfiguration` to customize the `ObjectMapper`. Does that do the trick for you?

**Comment by vpavic** [comment]
The originally reported problem disappeared for us when we moved to Spring HATEOAS 1.0.

To clarify: we're also using a custom media type that's based on HAL, and with 1.0 providing a first-class support custom media types, IIRC we were able to remove the BPP-based workaround by simply implementing `HypermediaMappingInformation#configureObjectMapper`.

For stock HAL usage I do find these new callbacks quite helpful and will certainly make use of them so many thanks for this addition. :slightly_smiling_face:

So I assume this issue can be closed, right?
